perllib/FixMyStreet/App/Controller/Dashboard.pm

```diff
@@ -77,141 +69,219 @@ sub check_page_allowed : Private {
+        my $children = mySociety::MaPit::call('area/children', $area_id,
+            type => $c->cobrand->area_types_children,
+        );
+        $c->stash->{children} = $children;
 
-    my $council_detail = mySociety::MaPit::call('area', $area_id );
-    $c->stash->{council} = $council_detail;
+        $c->forward('/admin/fetch_contacts');
+        $c->stash->{contacts} = [ $c->stash->{contacts}->all ];
 
-    my $children = mySociety::MaPit::call('area/children', $area_id,
-        type => $c->cobrand->area_types_children,
-    );
-    $c->stash->{children} = $children;
+        # See if we've had anything from the body dropdowns
+        $c->stash->{category} = $c->get_param('category');
+        $c->stash->{ward} = $c->get_param('ward');
+        if ($c->user->area_id) {
+            $c->stash->{ward} = $c->user->area_id;
+        }
+    } else {
+        $c->forward('/admin/fetch_all_bodies');
+    }
 
-    $c->stash->{all_areas} = { $area_id => $council_detail };
-    $c->forward( '/report/new/setup_categories_and_bodies' );
+    $c->stash->{start_date} = $c->get_param('start_date');
+    $c->stash->{end_date} = $c->get_param('end_date');
+    $c->stash->{q_state} = $c->get_param('state') || '';
 
-    # See if we've had anything from the dropdowns
+    $c->forward('construct_rs_filter');
 
-    $c->stash->{ward} = $c->get_param('ward');
-    $c->stash->{category} = $c->get_param('category');
+    if ( $c->get_param('export') ) {
+        $self->export_as_csv($c);
+    } else {
+        $self->generate_data($c);
+    }
+}
 
-    my %where = (
-        'problem.state' => [ FixMyStreet::DB::Result::Problem->visible_states() ],
-    );
+sub construct_rs_filter : Private {
+    my ($self, $c) = @_;
+
+    my %where;
     $where{areas} = { 'like', '%,' . $c->stash->{ward} . ',%' }
         if $c->stash->{ward};
     $where{category} = $c->stash->{category}
         if $c->stash->{category};
-    $c->stash->{where} = \%where;
-    my $prob_where = { %where };
-    $prob_where->{'me.state'} = $prob_where->{'problem.state'};
-    delete $prob_where->{'problem.state'};
-    $c->stash->{prob_where} = $prob_where;
 
-    my $dtf = $c->model('DB')->storage->datetime_parser;
-
-    my %counts;
-    my $now = DateTime->now( time_zone => FixMyStreet->local_time_zone );
-    my $t = $now->clone->truncate( to => 'day' );
-    $counts{wtd} = $c->forward( 'updates_search',
-        [ $dtf->format_datetime( $t->clone->subtract( days => $t->dow - 1 ) ) ] );
-    $counts{week} = $c->forward( 'updates_search',
-        [ $dtf->format_datetime( $now->clone->subtract( weeks => 1 ) ) ] );
-    $counts{weeks} = $c->forward( 'updates_search',
-        [ $dtf->format_datetime( $now->clone->subtract( weeks => 4 ) ) ] );
-    $counts{ytd} = $c->forward( 'updates_search',
-        [ $dtf->format_datetime( $t->clone->set( day => 1, month => 1 ) ) ] );
+    my $state = $c->stash->{q_state};
+    if ( $state eq 'fixed - council' ) {
```

Comment:
I'm a bit wary about this one fixed state having special status in here biting us in the future. Would it not make more sense for all fixed states to behave like this?

Reply:
Yes, that is its purpose - the drop down in the front end always has 'fixed - council' because it's the staff user dropdown, and if they pick that option, this is saying map it to all fixed states in the lookup.

Reply:
The `fixed - council` thing in the dropdown is becuase that's whats listed in `state_groups_inspect` by default. I was just thinking about if someone changed that in a cobrand. Possibly I am thinking too hard about this :)

Reply:
Ah, if they did that, they'd be very silly :) Whilst you can rename the fixed state in the admin, you can't have multiple ones, it's only a tickbox that creates a 'fixed' state, and if that's present, the code sticks in 'fixed - user' and 'fixed - council' too. So all you could possibly change it to is 'fixed' or 'fixed - user', which wouldn't make much sense. But yes, it could check fixed_states, why not.